doc/specs/tagging_technical.md

```diff
@@ -0,0 +1,116 @@
+    * A pre-defined tag list, or null to use the global tag list
+    * A lower tag limit (-1 for no limit)
+    * An upper tag limit (-1 for no limit)
+    * A boolean indicating if voting is enabled or disabled (true for enabled, false for disabled)
+  * Second constructor will create a heirarchal tagging instance, accepting the following
+    parameters:
+    * The item name (i.e. thread, wiki article), used for schema
+    * The tag editing permission name. This will vary based on the system utilizing
+      tagging
+    * A boolean indicating if parent tags are required or optional
+    * A boolean indicating if child tags are required or optional
+    * A boolean indicating if parent tags must be chosen before child tags
+    * A pre-defined parent tag list, or null to use the global tag list
+    * A pre-defined child tag list, or null to use the global tag list
+    * The lower parent tag limit (-1 for no limit)
+    * The upper parent tag limit (-1 for no limit)
+    * The lower child tag limit (-1 for no limit)
+    * The upper child tag limit (-1 for no limit)
+    * A boolean indicating if voting is enabled or disabled (true for enabled, false for disabled)
+  * All parameters must be stored as local data for reuse in functions.
+  * Function: addTag(item_id, tag_id) [simple]
+    * Lookup tag name, and insert the data into %item%_tags if not blacklisted
+  * Function: addTag(item_id, tag_id, parent_tag) [heirarchal]
+    * parent_tag: bool, indicating if the tag is a parent tag
+    * Lookup tag name, and insert the data into %item%_tags if not blacklisted
+  * Function: removeTag(item_id, tag_id) [both]
+    * Remove corresponding row from %item%_tags
+  * Function: upvoteTag(item_id, tag_id) [both]
+    * Update %item%_tags corresponding row: votes = votes + 1
+  * Function: downvoteTag(item_id, tag_id) [both]
+    * Update %item%_tags corresponding row: votes = votes - 1
+* Class: GlobalTagManager
+  * Constructor takes no arguments.
+  * Function: addTag(tag_name)
+    * Add a tag to the global taglist
+  * Function: removeTag(tag_id)
+    * Remove a tag from the global taglist
+  * Function: blacklistTag(tag_name)
+    * Add a tag to the global blacklist
+  * Function: removeTagFromBlacklist(tag_id)
+    * Remove a tag from the global blacklist
+  * Function: renameTag(tag_id, new_name, type)
+    * Rename a tag
+    * Type is a string (whitelist or blacklist), indicating if the tag is on the whitelist or blacklist
+
+## Data Model
+
+* %item%_tags:
+    * %item%_id: int, the item ID (thread ID, wiki article ID, etc.)
+    * tag: string, the tag name
+    * tag_id: string, the tag ID
+    * parent_tag: boolean, only created if heirarchal tagging is chosen. true if parent tag,
+      false if child tag
+    * votes: int, only created if voting is enabled. Contains the net number of votes
+    * added_by: string, the ID of the user that added the tag
+    * added_at: string, the timestamp of when the tag was added
```

Comment:
Following the meteor style guide and the mongo guidelines that will be merged, we want collections, fields, and variable identifiers to be camelCased.